sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -75,6 +92,28 @@ def test_partial_dependence_regressor():
     assert pdp.shape == (1, grid_resolution)
     assert axes[0].shape[0] == grid_resolution
 
+    # Test near perfect correlation between partial dependence and diagonal
```

Comment:
Great test, can you please make it a stand-alone `test_sample_weight()`? Would be easier for me to port ;)

Reply:
Indeed, this could be reused down the line, done !